### PR TITLE
coverity fixes for High priority failures

### PR DESCRIPTION
**Pull request by rileyberton**

### CODE REVIEW

File: src/modules/dns.c [hunk 1]
```diff
@@ -484,7 +484,7 @@ static void dns_module_eventer_dns_utm_fn(struct dns_ctx *ctx,
     dns_module_dns_ctx_release(h); /* acquire happened on timeout assigment */
   }
 
-  if(ctx == NULL) {
+  if(ctx == NULL || h == NULL) {
```
Comment (rileyberton) [inline src/modules/dns.c]: https://scan7.coverity.com/reports.htm#v20002/p12029/fileInstanceId=30445813&defectInstanceId=6775566&mergedDefectId=1452416

File: src/modules/histogram.c [hunk 2]
```diff
@@ -70,7 +70,7 @@ static int
 histogram_config(mtev_dso_generic_t *self, mtev_hash_table *o) {
   struct histogram_config *conf = mtev_image_get_userdata(&self->hdr);
   const char *duty;
-  char *duty_copy, *cp, *brk;
+  char *duty_copy, *cp, *brk = NULL;
```
Comment (rileyberton) [inline src/modules/histogram.c]: https://scan7.coverity.com/reports.htm#v20002/p12029/fileInstanceId=30445825&defectInstanceId=6775632&mergedDefectId=1452419

File: src/modules/postgres_ingestor.c [hunk 3]
```diff
@@ -1329,6 +1329,7 @@ storage_node_quick_lookup(const char *uuid_str, const char *remote_cn,
         if(info->dsn) free(info->dsn);
         if(info->fqdn) free(info->fqdn);
         free(info);
+        info = NULL;
```
Comment (rileyberton) [inline src/modules/postgres_ingestor.c]: https://scan7.coverity.com/reports.htm#v20002/p12029/fileInstanceId=30445924&defectInstanceId=6775613&mergedDefectId=1452413

File: src/modules/snmp.c [hunk 4]
```diff
@@ -239,58 +239,58 @@ static int noit_snmp_accumulate_results(noit_check_t *check, struct snmp_pdu *pd
   struct check_info *info = check->closure;
   struct variable_list *vars;
 
-  if(pdu)
+  if(pdu) {
```
Comment (rileyberton) [inline src/modules/snmp.c]: https://scan7.coverity.com/reports.htm#v20002/p12029/fileInstanceId=30445840&defectInstanceId=6775565&mergedDefectId=1452418&fileStart=251&fileEnd=500

File: src/noit_check_log_helpers.c [hunk 5]
```diff
@@ -76,9 +76,11 @@ noit_check_log_bundle_compress_b64(noit_compression_type_t ctype,
       break;
     }
   case NOIT_COMPRESS_NONE:
-    compbuff = (char *)buf_in;
-    break;
-  }
+    {
```
Comment (rileyberton) [inline src/noit_check_log_helpers.c]: https://scan7.coverity.com/reports.htm#v20002/p12029/fileInstanceId=30445737&defectInstanceId=6775611&mergedDefectId=1452405

File: src/noit_clustering.c [hunk 6]
```diff
@@ -377,10 +377,10 @@ static CURL *get_curl_handle() {
 }
 static size_t write_data_to_file(void *buff, size_t s, size_t n, void *vd) {
   int *fd = vd;
-  size_t len = s*n;
+  ssize_t len = s*n;
   mtevL(cldeb, "XML[%.*s]\n", (int)len, (char *)buff);
-  while((len == write(*fd, buff, len)) == -1 && errno == EINTR);
-  return len;
+  while((len = write(*fd, buff, (size_t)len)) == -1 && errno == EINTR);
```
Comment (rileyberton) [inline src/noit_clustering.c]: https://scan7.coverity.com/reports.htm#v20002/p12029/fileInstanceId=30445778&defectInstanceId=6775549&mergedDefectId=1452415

File: src/noit_metric.c [hunk 7]
```diff
@@ -160,6 +160,7 @@ noit_metric_to_json(noit_metric_message_t *metric, char **json, size_t *len, mte
     if (field != NULL) {
       mtev_json_object_object_add(o, "status", mtev_json_object_new_string(field));
     }
+    free(status);
```
Comment (rileyberton) [inline src/noit_metric.c]: https://scan7.coverity.com/reports.htm#v20002/p12029/fileInstanceId=30445799&defectInstanceId=6775597&mergedDefectId=1376794

**Comment by pamaddox** [comment]
LGTM

**Pull request merged**